File: python/selenium/python2.7/iOSWebTest.py
from selenium import webdriver
from selenium.webdriver.remote.remote_connection import RemoteConnection
import unittest
import urllib3
urllib3.disable_warnings()
import sys
sys.path.append('..')
import configs

class iOSWebTest(unittest.TestCase):

  wrong_username_msg = 'Your username is invalid!'
  wrong_password_msg = 'Your password is invalid!'
  success_msg = 'You logged into a secure area!'

  def setUp(self):
    self._command_executor = RemoteConnection(configs.kobitonServerUrl, resolve_ip=False)
    self._command_executor.set_timeout(configs.session_timeout)
    self.driver = webdriver.Remote(configs.kobitonServerUrl, configs.desired_caps_ios_web)
    self.driver.implicitly_wait(configs.session_timeout)
    self.driver.set_page_load_timeout(configs.session_timeout)

    kobitonSessionId = self.driver.desired_capabilities.get('kobitonSessionId')
    print("https://portal.kobiton.com/sessions/%s" % (kobitonSessionId))

  def tearDown(self):
    self.driver.quit()

  def test_ios_web(self):
    self.verify_invalid_username()
    self.verify_invalid_password()
    self.verify_login_successfully()

  def verify_invalid_username(self):
    print 'should return error when we input wrong username'
    self.login('foo', 'SuperSecretPassword!')
    self.assertTrue(self.wrong_username_msg in self.get_message())

  def verify_invalid_password(self):
    print 'should return error when we input wrong password'
    self.login('tomsmith', 'SuperSecretPassword')
    self.assertTrue(self.wrong_password_msg in self.get_message())

  def verify_login_successfully(self):
    print 'should run test successfully with correct username and password'
    self.login('tomsmith', 'SuperSecretPassword!')
    self.assertTrue(self.success_msg in self.get_message())

  def login(self, username, password):
    self.driver.get('http://the-internet.herokuapp.com/login')
    userEle = self.driver.find_element_by_xpath(".//*[@id='username']")
    userEle.send_keys(username)
    self.driver.find_element_by_id('password').send_keys(password)
    self.driver.find_element_by_xpath("//form[@name='login']").submit()

  def get_message(self):
    return self.driver.find_element_by_id('flash').text

if __name__ == '__main__':
  suite = unittest.TestLoader().loadTestsFromTestCase(iOSWebTest)
  unittest.TextTestRunner(verbosity=2).run(suite)

File: ruby_rspec/iosWebTest.rb
require "selenium-webdriver"
require_relative "configs" 

describe "iOS Web Test" do

  before(:all) do
    caps = DesiredCapabilities.new
    @driver = Selenium::WebDriver.for(:remote,
      :url => caps.desired_caps_ios_web[:url],
      :desired_capabilities => caps.desired_caps_ios_web[:caps])
  end
    
  after(:all) do
    @driver.quit
  end 

  it "wrong username" do
    wrong_username_msg = 'Your username is invalid!'
    puts 'Test case 1: it should return error when we input wrong username'
    login('foo', 'SuperSecretPassword!')
    expect(get_message).to include(wrong_username_msg)
  end

  it "wrong password" do
    wrong_password_msg = 'Your password is invalid!'
    puts 'Test case 2: it should return error when we input wrong password'
    login('tomsmith', 'SuperSecretPassword')
    expect(get_message).to include(wrong_password_msg)
  end

  it "login successfully" do
    success_msg = 'You logged into a secure area!'
    puts 'Test case 3: it should run test successfully with correct username and password'
    login('tomsmith', 'SuperSecretPassword!')
    expect(get_message).to include(success_msg)
  end
end

def login(username, password)
  sleep(5)
  @driver.navigate.to('http://the-internet.herokuapp.com/login')
  @driver.find_element(:id, 'username').send_keys(username)
  @driver.find_element(:id, 'password').send_keys(password)
  @driver.find_element(:xpath, "//form[@name='login']").submit
end

def get_message
  return @driver.find_element(:id, 'flash').text
end


